weasyl/controllers/settings.py

```diff
@@ -52,34 +54,53 @@ def control_editprofile_get_(request):
 @login_required
 @token_checked
 def control_editprofile_put_(request):
-    form = request.web_input(
-        full_name="", catchphrase="",
-        profile_text="", set_commish="", set_trade="", set_request="",
-        set_stream="", stream_url="", stream_text="", show_age="",
-        gender="", country="", profile_display="", site_names=[], site_values=[])
+    site_names = request.params.getall('site_names')
+    site_values = request.params.getall('site_values')
+    ftrade = request.params.get('set_trade', '')
+    frequest = request.params.get('set_request', '')
+    fcommish = request.params.get('set_commish', '')
 
-    if len(form.site_names) != len(form.site_values):
+    if len(site_names) != len(site_values):
         raise WeasylError('Unexpected')
 
-    if 'more' in form:
-        form.username = define.get_display_name(request.userid)
-        form.sorted_user_links = [(name, [value]) for name, value in zip(form.site_names, form.site_values)]
-        form.settings = form.set_commish + form.set_trade + form.set_request
-        form.config = form.profile_display
-        return Response(define.webpage(request.userid, "control/edit_profile.html", [form, form], title="Edit Profile", options=["typeahead"]))
+    sites = zip(site_names, site_values)
+
+    if 'more' in request.params:
```

Comment:
```suggestion
    # Allow adding additional links to the profile via 'New' button without intermediate save if JavaScript is disabled
    if 'more' in request.params:
```
The only comment I have here is we might consider a comment here... it took me _way_ too long to grok the code and realize that this ``more`` parameter typically isn't present; it only is if scripts are off, for adding new links without actually saving in-between each newly added entry until done.